### ISSUE TITLE
Error: No such file or directory (os error 2)

### ISSUE THREAD

**Issue by johnoneil** [mozilla/grcov#656]
Surprised I'm not seeing this issue reported but there must be something fundamentally different about our use case that is bringing out a very odd issue. We're working in Rust, trying to get an `html` coverage report.

```
rustc --version
rustc 1.53.0 (53cb7b09b 2021-06-17)
grcov --version
grcov 0.8.0
```
Our project is a workspace, which may be (?) the source of the issue.

Following your `README` we generate numerous `.profraw` files, all located at the root of the sub-crates within the workspace, not at the workspace root. Does this seem incorrect?

Then running `grcov` as follows:

```
grcov . --binary-path ./target/debug/ --source-dir ./ -t html --branch --ignore-not-existing --ignore "*cargo*" -o ./coverage/
```
But get an error `No such file or directory (os error 2)`.

The specific error is long but here's the highlights...

```
21:27:44 [ERROR] Error while executing llvm tools: Failed to execute "/home/joneil/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-profdata" "merge" "-sparse" "/tmp/.tmppfF8bO/crates/toolkit/coverage-112923-6803954054635160813_0_1.profraw"
...
"-o" "/tmp/.tmppfF8bO/0/grcov.profdata"
No such file or directory (os error 2)
```

First there's no message about which (of many hundreds of files) is missing. But none of the files dumped in the messge are present. Our `.profraw` files are in each crate, not in `/tmp/.

Where there are no files at `/tmp/.tmppfF8b0`. Ought the `profraw` files be there rather than spread throughout our workspace? Very confused by the expectation of `.profdata` files being in the `/tmp/` directory.

Any help is appreciated.

**Comment by marco-c** [comment]
grcov is creating symlinks to the profraw files in the tmp directory (for historical reasons, as it was needed for gcno/gcda files to run in parallel). https://github.com/mozilla/grcov/blob/89cb517997fbb2e8f8177b22e0be0b183ba7671a/src/producer.rs#L408

So, the temporary directory should exist while grcov is running (as grcov is creating it). IIRC it is deleted at the end of the grcov execution, so that's why the directory is empty when you go check it.

There must be some other problem.

Can you try with Nightly instead of stable? I see you're trying to use the llvm-tools from stable (`/home/joneil/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-profdata`).

**Comment by johnoneil** [comment]
Will do. Thanks for your input @marco-c .

I think it's likely I'll be debugging this this week so if/when I can get it working I'll post here to close.

**Comment by johnoneil** [comment]
I was able to debug this today.

The original of my issue was not using rust nightly, as @marco-c suggested, but also not installing the `llvm-tools-preview` component specifically for nightly. I don't yet have a good workflow for this configuration but that's not related to this issue.

Running into issue #555 (which I don't believe has found its way into release yet) but I was able to get coverage stats using `grcov` on current master.

Closing.
